### PR TITLE
chore: update dependencies and fix eslint errors

### PR DESCRIPTION
**Pull request by aromko**

### CODE REVIEW

File: src/routes/state-management/_components/ServerStateExample.tsx [hunk 1]
```diff
@@ -59,6 +60,7 @@ function ServerStateExample() {
             width={'1/5'}
             onChange={key => {
               navigate({
+                // @ts-ignore
```
Comment (sarahgm) [inline src/routes/state-management/_components/ServerStateExample.tsx:63]: why did you add these comments?

**Review by sarahgm — CHANGES_REQUESTED** [review]

File: src/routes/state-management/_components/ServerStateExample.tsx (hunk 1 above)
Comment (aromko) [reply src/routes/state-management/_components/ServerStateExample.tsx:63]: Because I red the tanstack documentation which provides examples that looks exactly the same. But typescript moans about it that it doesn't exists and I didn't find out why. And at some point I thought I don't care. I also don't like it - and you shouldn't do it - but if you have an idea feel free to change it.

Comment (aromko) [reply src/routes/state-management/_components/ServerStateExample.tsx:63]: https://tanstack.com/router/v1/docs/framework/react/guide/search-params#usenavigate-navigate-search-

Comment (aromko) [reply src/routes/state-management/_components/ServerStateExample.tsx:63]: Btw I just hope it's goes away in another version^^

**Review by sarahgm — APPROVED** [review]
